moveit_core/collision_detection/include/moveit/collision_detection/world.h

```diff
@@ -91,6 +91,13 @@ class World
      *  This frame is returned when getTransform() is called with the object's name. */
     Eigen::Isometry3d pose_;
 
+    /** \brief (Optional) The URL of the mesh used for the visual representation of this object.
+     * Only used for display in Rviz, not for collision checking. */
+    std::string visual_geometry_mesh_url_;
+
+    /** \brief The pose of the visual geometry (mesh). */
+    Eigen::Isometry3d visual_geometry_pose_;
```

Comment:
```suggestion
    Eigen::Isometry3d visual_geometry_pose_ = Eigen::Isometry3d::Identity();
```
then you can skip that later

Reply:
If this creates issues with `Eigen::Isometry3d` being forward declared (but I don't think it is ...) you could also set the value in the constructor and move the constructor to the cpp file (where it actually belongs ...)